command/src/scm_socket.rs

```diff
@@ -20,7 +20,7 @@ pub const MAX_BYTES_OUT: usize = 4096;
 #[derive(Clone, Debug, Serialize, Deserialize)]
 pub struct ScmSocket {
     pub fd: RawFd,
-    pub blocking: bool,
+    pub blocking: bool, // TODO this is never used anywhere. better remove it
```

Comment:
We should populate it and check the value, when using `set_blocking` as under the hood it performs a `syscall` which took time to complete. So, this is more likely a regression that an unused field

Reply:
I agree. Then again modifying `set_blocking` in order to update the `blocking` field, means changing its signature to take `&mut self` instead of `&self` as an argument, which hinders its use in the `server` module.

I'll just leave the comment and write an entire issue about this: #841

Reply:
I think, it is not a big deal to change te immutable reference to self to a mutable one